synapse/cores/common.py

```diff
@@ -259,67 +297,54 @@ def setModlVers(self, name, vers):
             self.setRowsByIdProp(iden, prop, vers)
             return vers
 
-    def revModlVers(self, name, revs):
-        '''
-        Update a model using a list of (vers,func) tuples.
+    def _addDataModels(self, modtups):
 
-        Args:
-            name (str): The name of the model
-            revs ([(int,function)]):  List of (vers,func) revision tuples.
+        DataModel._addDataModels(self, modtups)
 
-        Returns:
-            (None)
-
-        Example:
+        for name, modl in modtups:
 
-            with s_cortex.openurl('ram:///') as core:
+            for tnam, tnfo in modl.get('types', ()):
+                self.addRuntNode('syn:type', tnam, **tnfo)
 
-                def v0():
-                    addModelStuff(core)
+            for fnam, fnfo, props in modl.get('forms', ()):
 
-                def v1():
-                    addMoarStuff(core)
+                self.addRuntNode('syn:form', fnam, **fnfo)
+                self.addRuntNode('syn:prop', fnam, **fnfo)
 
-                revs = [ (0,v0), (1,v1) ]
+                for pnam, pnfo in props:
+                    full = fnam + ':' + pnam
+                    self.addRuntNode('syn:prop', full, **pnfo)
 
-                core.revModlVers('foo', revs)
-
-        Each specified function is expected to update the cortex including data migration.
+    def revModlVers(self, name, vers, func):
         '''
-        if not revs:
-            return
-
-        curv = self.getModlVers(name)
+        Update and track a model version using a callback function.
 
-        maxver = revs[-1][0]
-        if maxver == curv:
-            return
+        Args:
+            name (str): The name of the model
+            vers (int): The version int ( in YYYYMMDDHHMM int format )
+            func (function):  The version update function
 
-        if not self.getConfOpt('rev:model'):
-            raise s_common.NoRevAllow(name=name, mesg='add rev:model=1 to cortex url to allow model updates')
+        Returns:
+            (None)
 
-        for vers, func in sorted(revs):
 
-            if vers <= curv:
-                continue
+        Each specified function is expected to update the cortex including data migration.
+        '''
+        if not self.getConfOpt('rev:model'):
+            raise s_common.NoRevAllow(name=name, vers=vers)
```

Comment:
Do we want to raise an exception here (and cause __init__ to bail) or just return?

Reply:
yea, we want to go down in a ball of flames here to prevent using a newer version of the data model without the approval of the configuration.  Otherwise, we'd end up silently creating nodes with a newer model and end up with *some* nodes in the newer configuration and others not ( a migration *nightmare* )

Reply:
:+1: